checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsVisitor.java

```diff
@@ -0,0 +1,85 @@
+package org.checkerframework.checker.calledmethods;
+
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.calledmethods.framework.FrameworkSupport;
+import org.checkerframework.checker.calledmethods.qual.CalledMethods;
+import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey;
+import org.checkerframework.common.accumulation.AccumulationVisitor;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.TreeUtils;
+
+/**
+ * This visitor implements the custom error message finalizer.invocation.invalid, and supports
+ * counting the number of framework build calls.
+ */
+public class CalledMethodsVisitor extends AccumulationVisitor {
+
+    /**
+     * Custom error message key for finalizer invocations. The custom error message makes it easier
+     * for users to tell what methods weren't called.
+     */
+    public static final @CompilerMessageKey String FINALIZER_INVOCATION_INVALID =
+            "finalizer.invocation.invalid";
+
+    /**
+     * Constructor matching super.
+     *
+     * @param checker the type-checker associated with this visitor
+     */
+    public CalledMethodsVisitor(final BaseTypeChecker checker) {
+        super(checker);
+    }
+
+    @Override
+    public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
+
+        if (checker.getBooleanOption(CalledMethodsChecker.COUNT_FRAMEWORK_BUILD_CALLS)) {
+            ExecutableElement element = TreeUtils.elementFromUse(node);
+            for (FrameworkSupport frameworkSupport :
+                    ((CalledMethodsAnnotatedTypeFactory) getTypeFactory()).getFrameworkSupports()) {
+                if (frameworkSupport.isBuilderBuildMethod(element)) {
+                    ((CalledMethodsChecker) checker).numBuildCalls++;
+                    break;
+                }
+            }
+        }
+        return super.visitMethodInvocation(node, p);
+    }
+
+    /**
+     * Adds special reporting for method.invocation.invalid errors to turn them into
```

Comment:
I see no transformation that turns some errors into others.

Reply:
It is implicit: this method in `BaseTypeVisitor` normally would report a `method.invocation.invalid` error every time it is called. This implementation doesn't do that, and instead reports `finalizer.invocation.invalid` errors in the same places.